### PR TITLE
[MIG] Master google spreadsheet new api vbh

### PR DESCRIPTION
**Pull request by vpulbhatt**
- Task : https://www.odoo.com/web#id=12148&view_type=form&model=project.task&menu_id=3942&action=327
- Pad : https://pad.odoo.com/p/r.dF6IczDDl4hnZHMg

### CODE REVIEW

File: addons/google_spreadsheet/__openerp__.py [hunk 1]
```diff
@@ -7,17 +7,16 @@
...
-        'google_spreadsheet_data.xml',
+        'views/google_spreadsheet_view.xml',
+        'data/google_spreadsheet_data.xml',
+        'views/google_sheets_req_format_templates.xml',
         'views/google_spreadsheet.xml',
     ],
-    'qweb': ['static/src/xml/*.xml'],
-    'demo': [],
+    'qweb': ['static/src/xml/addtospreadsheet.xml'],
     'installable': True,
```
Comment (msh-odoo) [inline addons/google_spreadsheet/__openerp__.py]: installable is by default True

File: addons/google_spreadsheet/models/google_spreadsheet.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        return True
+
+    @api.model
+    def set_spreadsheet(self, model, domain, groupbys, view_id):
+        try:
+            google_drive_config = self.env.ref('google_spreadsheet.google_spreadsheet_template')
+        except ValueError:
+            raise
+        title = _('Spreadsheet %s' % model)
+        res = self._model.copy_doc(self.env.cr, self.env.uid, False,
```
Comment (msh-odoo) [inline addons/google_spreadsheet/models/google_spreadsheet.py]: Humm, It seems issue with copy_doc method, which is decorated with cr_uid_id_context, that's why you called it like ```self._model.copy_doc ```
Shouldn't we fix it in copy_doc method itself(may be in separate commit).

**Comment by msh-odoo** [comment]
==> domain on action action_ir_attachment_google_spreadsheet_tree should be [('url', 'like', '/spreadsheets/')]
==> Instead of defining action_ir_attachment_google_spreadsheet_tree_view and action_ir_attachment_google_spreadsheet_form_view, we may directly create view_ids on action and assign reference of views(tree and form).
==> Rebase your branch, you should get conflicts, we replaced simplejson with json

File: addons/google_spreadsheet/models/google_spreadsheet.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import cgi
+from lxml import etree
+import logging
+import simplejson
+import urllib2
+import werkzeug.urls
```
Comment (msh-odoo) [inline addons/google_spreadsheet/models/google_spreadsheet.py]: Unused import

File: addons/google_spreadsheet/models/google_drive_config.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        if attachment_id:
+            self.env['ir.attachment'].browse(attachment_id).write({'description': description})
+        return True
+
+    @api.model
+    def set_spreadsheet(self, model, domain, groupbys, view_id):
+        try:
+            google_drive_config = self.env.ref('google_spreadsheet.google_spreadsheet_template')
+        except ValueError:
+            raise
```
Comment (msh-odoo) [inline addons/google_spreadsheet/models/google_drive_config.py]: If ValueError is raised then raise it, does it make sense ? xmlid_to_object of ir.model.data throwing ValueError(With user friendly message).

File: addons/google_drive/google_drive.py [hunk 5]
```diff
@@ -67,7 +67,7 @@ def get_access_token(self, cr, uid, scope=None, context=None):
             req = urllib2.Request('https://accounts.google.com/o/oauth2/token', data, headers)
             content = urllib2.urlopen(req, timeout=TIMEOUT).read()
         except urllib2.HTTPError:
-            if user_is_admin:
+            if self.pool['res.users']._is_admin(cr, uid, [uid]):
```
Comment (hmo-odoo) [inline addons/google_drive/google_drive.py]: plz do not change in other module

File: addons/google_spreadsheet/models/google_drive_config.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            formula = '=oe_browse("%s";"%s";"%s")' % (model, fields, domain)
+
+        url = self.env['ir.config_parameter'].get_param('web.base.url')
+        if not self.env.user.password:
+            config_formula = '=oe_settings("%s";"%s")' % (url, self.env.cr.dbname)
+        else:
+            config_formula = '=oe_settings("%s";"%s";"%s";"%s")' % (url, self.env.cr.dbname, self.env.user.login, self.env.user.password)
+
+        template = self.env.ref('google_spreadsheet.sheets_api_request_format')
+        feed = etree.fromstring(template.arch_base).findall('{http://www.w3.org/2005/Atom}feed')[0]
```
Comment (hmo-odoo) [inline addons/google_spreadsheet/models/google_drive_config.py]: plz make separate commit to refector this method to easy understand why need it.

**Comment by jem-odoo** [comment]
Merged at https://github.com/odoo/odoo/commit/2fcf05d6d8a79db473c75bc91a43b6afbe125edc
